Project/DAQ_System/Program/Src/Sensor/Accelerometer/LSM6DSOXAccelerometer.hpp

```diff
@@ -28,33 +28,34 @@
 
 namespace sensor{
 
-class LSM6DSOX: public IAccelerometer {
+namespace AccelerometerConfiguration{
+	enum class ODR : uint8_t{
+		ODR12_5  = 0x01,
+		ODR26    = 0x02,
+		ODR52    = 0x03,
+		ODR104   = 0x04,
+		ODR208   = 0x05,
+		ODR416   = 0x06,
+		ODR833   = 0x07,
+		ODR1_66K = 0x08,
+		ODR3_33K = 0x09,
+		ODR6_66K = 0x0A
+	};
+
+	enum class FSR : uint8_t{
+		FSR2g  = 0x00,
+		FSR4g  = 0x02,
+		FSR8g  = 0x03,
+		FSR16g = 0x1
+	};
+}
```

Comment:
I think it makes more sense if this is nested inside the class declaration. If you're worried the scoping being too long, you can remove `Accelerometer` from `AccelerometerConfiguration`. So then it will look like `sensor::LSM6DSOX::Configuration::ODR::value`.

Either way, the consumer of your library can shorten or rename this as they please. For example:

```c++
using AccelerometerODR = sensor::LSM6DSOX::Configuration::ODR;

int main() {
  AccelerometerODR odr_mode = AccelerometerODR::ODR26;
}
```

or

```c++
using ODR = sensor::LSM6DSOX::Configuration::ODR;

int main() {
  accelerometer->SetODR(ODR::ODR26);
}
```

Reply:
Yeah it do make more sense to have it in the class